azdev/operations/extensions/metadata.py

```diff
@@ -0,0 +1,515 @@
+
+        for key, value in sorted(may_requires.items(), key=sort_key):
+            may_requirement = OrderedDict((('requires', value),))
+            # may_requirement = defaultdict((('requires', value),))
+            if key.extra:
+                may_requirement['extra'] = key.extra
+            if key.condition:
+                may_requirement['environment'] = key.condition
+            metadata['run_requires'].append(may_requirement)
+
+        if 'extras' not in metadata:
+            metadata['extras'] = []
+        metadata['extras'].extend([key.extra for key in may_requires.keys() if key.extra])
+
+
+def get_wheel_generator(whl_path):
+    """
+    Extract the Generator value from the WHEEL file in a .whl package.
+
+    Args:
+        whl_path (str): Path to the .whl file.
+
+    Returns:
+        str: The Generator value, or None if not found.
+    """
+    try:
+        # Ensure the file is a valid zip file
+        if not zipfile.is_zipfile(whl_path):
+            print(f"The file {whl_path} is not a valid zip file.")
+            return "bdist_wheel"
+
+        # Open the .whl file
+        with zipfile.ZipFile(whl_path, 'r') as zf:
+            # Locate the WHEEL file
+            wheel_file = None
+            for file_name in zf.namelist():
+                if ".dist-info/WHEEL" in file_name:
+                    wheel_file = file_name
+                    break
+
+            if not wheel_file:
+                print(f"WHEEL file not found in {whl_path}.")
+                return "bdist_wheel"
+
+            # Read and parse the WHEEL file
+            with zf.open(wheel_file) as wf:
+                for line in wf:
+                    decoded_line = line.decode("utf-8").strip()
+                    if decoded_line.startswith("Generator:"):
+                        _, generator_value = decoded_line.split(":", 1)
+                        return generator_value.strip()
+
+            print(f"Generator key not found in {wheel_file}.")
+            return "bdist_wheel"
+
+    except Exception as e:
+        print(f"An error occurred while processing {whl_path}: {e}")
+        return "bdist_wheel"
+
+
+def pkginfo_to_dict(path, distribution=None):
+    """
+    Convert PKG-INFO to a prototype Metadata 2.0 (PEP 426) dict.
+
+    The description is included under the key ['description'] rather than
+    being written to a separate file.
+
+    path: path to wheel file
+    distribution: optional distutils Distribution()
+    """
+    # metadata = DefaultDict(
+    #     lambda: DefaultDict( lambda: DefaultDict(defaultdict)))
+
+    metadata = OrderedDefaultDict(
+        lambda: OrderedDefaultDict(lambda: OrderedDefaultDict(OrderedDict)))
+
+    metadata["generator"] = "bdist_wheel (0.30.0)"
+    try:
+        pkg_info = read_pkg_info(path)
+    except Exception:
+        with open(path, 'rb') as pkg_info_file:
+            pkg_info = email.parser.Parser().parsestr(pkg_info_file.read().decode('utf-8'))
```

Comment:
Another more straightforward way to get pkgInfo is `pkginfo.Wheel` for `dist-info` or `pkginfo.Develop` for `egg-info`, as `az extension list` does in azure cli core here: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/extension/__init__.py#L154-L157